src/js/d3Utils.js

```diff
@@ -23,6 +23,27 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         return d3.selectAll(elem.toArray());
     };
 
+    /* Given the object "d", returns value of:
+     * - "id" if it exists on the top-level object,
+     * - "data.id" if it exists on a "data" object
+     * - "undefined" if neither of these exist
+     */
+    floe.d3.idExtractor = function (d) {
+        return d.id !== undefined ? d.id : d.data !== undefined ? d.data.id !== undefined ? d.data.id : undefined : undefined;
+    };
+
+    // Given a selection of D3 elements and an ID, returns only the elements
+    // matching that ID
+    floe.d3.filterById = function (d3Selection, idToFilter, invertResult) {
```

Comment:
The basic struggle I'm having in thinking about how best to implement this is the D3 piece. The basic "get DOM elements associated with the ID of this particular data point" functionality is enabled by D3 and its approach to binding information about data to DOM nodes and maintaining object constancy through keys. So as stands, at some point in the implementation we have to "ask" D3 to tell us which elements in the legend/pie match/don't match the ID associated with a data point.

That said, I'll hack on this further and see what I can come up with that limits the role of D3's paradigms further in favour of Infusion's.